mongo/mongo.go

```diff
@@ -711,14 +712,31 @@ func installMongod(mongoDep packaging.Dependency, hostSeries, dataDir string) er
 			EnableAtStartup: true,
 		},
 	}
-	conf := common.Conf{Desc: ServiceName + " snap"}
+	conf := newConf(&ConfigArgs{})
+	conf.Desc = ServiceName + " snap"
 	snapChannel := fmt.Sprintf("%s/%s", SnapTrack, SnapRisk)
-	service, err := snap.NewService(snapName, ServiceName, conf, snap.Command, snapChannel, "", backgroundServices, prerequisites)
+	snapSvc, err := snap.NewService(snapName, ServiceName, conf, snap.Command, snapChannel, "", backgroundServices, prerequisites)
 	if err != nil {
 		return errors.Trace(err)
 	}
 
-	return service.Install()
+	if usingMongoFromSnap {
+		defer func() {
+			err = snapSvc.ConfigOverride()
```

Comment:
Shouldn't we check if we already have an error here? (eg snapSvc.Install() returned an error, why would we do ConfigOverride?)